Auth added

Gleb1nho opened the pull request:
Добавил авторизацию. Можете заметить, что я тот ещё любитель спагетти, поэтому рефакторинг приветствуется:)

Gleb1nho committed:
Database should be working

Добавил модель пользователя и прописал маршруты для вывода всех/конкретного по githubid

Gleb1nho committed:
seems to remove unnecessary code

Добавил .next в .gitignore, конфиг бд теперь берётся из окружения

Gleb1nho committed:
Delete .next directory

Gleb1nho committed:
Auth added

Добавил авторизацию, сделал шаблон для вывода списка контактов
Код получился, мягко говоря, страшненький, но работает, рефакторинг приветствуется:)

Gleb1nho committed:
Update index.ts

Gleb1nho committed:
Update index.ts

Gleb1nho committed:
Update index.ts

Gleb1nho committed:
Update index.ts

Gleb1nho committed:
Update index.ts

Gleb1nho committed:
Update index.ts

Gleb1nho committed:
Update index.ts

Gleb1nho committed:
Add files via upload

Gleb1nho committed:
Fixed styles upload for hbs

Изменил в package.json папку, куда складываются стили, теперь всё должно нормально подключаться

Gleb1nho committed:
Merge branch 'auth-added' of https://github.com/urfu-2020/team-4 into auth-added

SofiSheromova commented on server/controllers/contacts.ts:
```diff
@@ -1,18 +1,36 @@
+    // Список всех пользователей
+    user.findAll().then(users => {
+        // eslint-disable-next-line prefer-const
+        let allUsers = [];
```
а почему не const всё-таки?

SofiSheromova commented on server/index.ts:
```diff
@@ -10,16 +11,115 @@ import 'isomorphic-fetch';
+    }
+};
+
+app.get('/', isAuth, (req, res) => {
+    res.redirect('/contacts');
+});
+
+app.get('/login', (req, res) => {
+    if (req.user) {
+        return res.redirect('/');
+    }
+    res.redirect('/auth/github');
+});
+
+app.get('/logout', (req, res) => {
+    req.logOut();
+    res.redirect('/login');
+});
+
+app.get('/auth/github',
+    passport.authenticate('github'));
+
+app.get('/auth/github/callback',
+    passport.authenticate('github', { failureRedirect: '/login' }),
+    function (req, res) {
+    // Successful authentication, redirect home.
+        res.redirect('/');
+    }
+);
```
для этого есть специальный файл

SofiSheromova commented on server/controllers/contacts.ts:
```diff
@@ -1,18 +1,36 @@
+        }
+        // eslint-disable-next-line no-console
+        console.log(allUsers);
+        res.render('contacts', { people: allUsers });
```
этого не будет, нужно просто отправлять json со списком пользователей

SofiSheromova replied:
а ещё api брался разрабатывать Ваня

Gleb1nho replied:
Руты убрал в файлы с рутами

Gleb1nho replied:
А авторизацию в файл с авторизацией

SofiSheromova reviewed (commented):
Нужно убрать hbs, у нас есть ограничения по технологиям на это задание
Перегрузил index.ts, можно оттуда вынести аутентификацию в отдельный файл/функции
Ну и роуты из index.ts тоже убрать

Gleb1nho committed:
Removed hbs

Убрал hbs, контакты отправляются просто json-ом.

SofiSheromova commented on package.json:
```diff
@@ -43,12 +43,22 @@
     "cross-env": "^7.0.3",
+    "dotenv": "^8.2.0",
     "express": "^4.17.1",
+    "express-handlebars": "^5.3.0",
```
не нужно

Gleb1nho replied:
Убрал handlebars

SofiSheromova commented on views/contacts.handlebars:
```diff
@@ -0,0 +1,12 @@
+{{#each contacts}}
```
тоже не нужно

SofiSheromova commented on views/layouts/main.handlebars:
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
и это не нужно

Gleb1nho committed:
HBS views directory removed

Gleb1nho committed:
Removed hbs from package.json

Gleb1nho committed:
Some refactoring for index.ts

Убрал ненужные вещи из index.ts в routes

Gleb1nho committed:
Updated db link

Поменял ссылку на бд, подходящей под модель старого формата. В будущем ссылка поменяется. Деплой должен ожить.

Gleb1nho committed:
Auth is independent now! :beers:

Убрал авторизацию в отдельный модуль

SofiSheromova committed:
correct work with next.js

SofiSheromova committed:
fix eslint

Gleb1nho committed:
Merge pull request #15 from urfu-2020/fix_next_express

Fix next-express

Gleb1nho committed:
Some refactoring
